packages/file_selector/file_selector/example/lib/main.dart

```diff
@@ -0,0 +1,298 @@
+class SaveTest extends StatefulWidget {
+  /// Constructor for the SaveTest page
+  SaveTest({Key key}) : super(key: key);
+
+  @override
+  _SaveTestState createState() => _SaveTestState();
+}
+
+class _SaveTestState extends State<SaveTest> {
+  final TextEditingController _fileController = TextEditingController();
+  final TextEditingController _nameController = TextEditingController();
+
+  @override
+  void dispose() {
+    _fileController.dispose();
+    super.dispose();
+  }
+
+  void _saveFile() async {
+    final path = await getSavePath();
+
+    final data = Uint8List.fromList(_fileController.text.codeUnits);
+
+    final new_file = (_nameController.text == '')
+        ? XFile.fromData(data, mimeType: 'text/plain')
+        : XFile.fromData(data,
+            mimeType: 'text/plain', name: _nameController.text);
+
+    new_file.saveTo(path);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Save Example"),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Container(
+              width: 300,
+              child: TextField(
+                minLines: 1,
+                maxLines: 12,
+                controller: _nameController,
+                decoration: InputDecoration(
+                  hintText: '(Optional) Suggest File Name',
+                ),
+              ),
+            ),
+            Container(
+              width: 300,
+              child: TextField(
+                minLines: 1,
+                maxLines: 12,
+                controller: _fileController,
+                decoration: InputDecoration(
+                  hintText: 'Enter File Contents',
+                ),
+              ),
+            ),
+            SizedBox(height: 10),
+            RaisedButton(
+              color: Colors.blue,
+              textColor: Colors.white,
+              child: Text('Press to save a text file'),
+              onPressed: () => {_saveFile()},
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// Screen that shows an example of openFile(s)
+class OpenTest extends StatefulWidget {
+  /// Default constructor
+  OpenTest({Key key}) : super(key: key);
+
+  @override
+  _OpenTestState createState() => _OpenTestState();
+}
+
+class _OpenTestState extends State<OpenTest> {
+  void _onOpenImageFile() async {
+    final typeGroup = XTypeGroup(label: 'images', extensions: ['jpg', 'png']);
+
+    final file = await openFile(acceptedTypeGroups: [typeGroup]);
+
+    await showDialog(
+        context: context,
+        builder: (context) {
+          return ImageDisplay(file: file);
+        });
+  }
+
+  void _onOpenTextFile() async {
+    final typeGroup = XTypeGroup(label: 'images', extensions: ['txt', 'json']);
+
+    final file = await openFile(acceptedTypeGroups: [typeGroup]);
```

Comment:
The only examples of `openFile` use exactly one accepted type group; there should ideally also be an example with none, and an example with multiple.

Reply:
I agree that we require more examples/ better documenttation for those particular cases. In that sense: 

- I revamped the example app to make the process of adding new examples easier. 
- I added a new example "open multiple images" that exercises the usage of multiple XTypeGroups

Given that this is going to be published as v0.7.0, leaving room for changes/improvements: Can we merge this as it currently is (so we can start with platform implementations) and add more examples/improve the docs as follow ups?

Reply:
We should be setting a good precedent in this plugin for what the examples platform implementations should cover. If it's easy to add examples, why not just do it now instead of hoping that someone will remember to come back and add it later?

Reply:
I added an example for `getDirectoryPath`. PTAL